### PR TITLE
Migrate services to getHashedData methods

### PR DESCRIPTION
**Pull request by stavamichal**
- migrate ad_mu and o365_mu to use GenHasedData class (in perl) instead
 of ServiceAttributes. Calling new method getHashedDataWithGroups.
 - for ad_mu: use virtual adDisplayName instead of defined one (for old
 behavior it should work the same way)

### CODE REVIEW

**Review by balcirakpeter — APPROVED** [review]
We have to decide what to do with the adDisplayName attribute. Otherwise, it seems to be ok.

**Comment by stavamichal** [comment]
If displayName is empty, we will use adName instead.

**Comment by stavamichal** [comment]
It should be fixed now.

**Review by balcirakpeter — APPROVED** [review]

File: gen/ad_mu [hunk 1]
```diff
@@ -518,19 +516,19 @@ for my $login (@logins) {
...
 
-	if($users->{$login}->{$A_U_F_O365_PREFERRED_LANGUAGE}) {
+	if ($users->{$login}->{$A_U_F_O365_PREFERRED_LANGUAGE}) {
 		print FILE "preferredLanguage: " . $users->{$login}->{$A_U_F_O365_PREFERRED_LANGUAGE} . "\n";
 	}
 
 	print FILE "targetaddress: " . $login . '@mail.muni.cz' . "\n";
 
-	print FILE "msDS-cloudExtensionAttribute1: " . $login . '@' . $facilityAttributes{$A_F_AZURE_DOMAIN} . "\n";
+	print FILE "msDS-cloudExtensionAttribute1: " . $login . '@' . $domain . "\n";
```
Comment (zlamalp) [inline gen/ad_mu:525]: ``$domain`` contains ``A_F_DOMAIN`` and not ``A_F_AZURE_DOMAIN`` after this change. Is it ok?

Comment (stavamichal) [reply gen/ad_mu]: Nice catch. You are right. Fixed.

Comment (stavamichal) [reply gen/ad_mu]: We didn't notice because there are the same values in both attributes at this moment in testing data.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**